### PR TITLE
Add msix emulation support

### PR DESCRIPTION
**Pull request by liuwei987**
-  hv: config: add msix emulation support
- acrn-config: generate msix emulation information

   Tracked-On: #4831
    Signed-off-by: Binbin Wu <binbin.wu@intel.com>
    Signed-off-by: Wei Liu <weix.w.liu@intel.com>
    Acked-by: Victor Sun <victor.sun@intel.com>
    Acked-by: Eddie Dong <eddie.dong@intel.com>

### CODE REVIEW

**Comment by binbinwu1** [comment]
@HuiHuangShi The MISRA-C check failed, but has no information of violations.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/arch/x86/configs/dnv-cb2/board.c [hunk 1]
```diff
@@ -23,3 +23,4 @@ struct platform_clos_info platform_mba_clos_array[MAX_PLATFORM_CLOS_NUM];
 
 const struct cpu_state_table board_cpu_state_tbl;
 const union pci_bdf plat_hidden_pdevs[MAX_HIDDEN_PDEVS_NUM];
+const struct vmsix_on_msi_dev_info vmsix_on_msi_devs[MAX_VMSIX_ON_MSI_PDEVS_NUM];
```
Comment (binbinwu1) [inline hypervisor/arch/x86/configs/dnv-cb2/board.c:26]: @liuwei987  Please help to update the struct name to 
struct vmsix_on_msi_info

File: hypervisor/arch/x86/configs/apl-up2/board.c [hunk 2]
```diff
@@ -48,3 +48,4 @@ const union pci_bdf plat_hidden_pdevs[MAX_HIDDEN_PDEVS_NUM] = {
 		.bits.f = 0x0,
 	},
 };
+const struct vmsix_on_msi_dev_info vmsix_on_msi_devs[MAX_VMSIX_ON_MSI_PDEVS_NUM];
```
Comment (binbinwu1) [inline hypervisor/arch/x86/configs/apl-up2/board.c:51]: @liuwei987  Please help to update the struct name to 
struct vmsix_on_msi_info

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**